### ISSUE TITLE
*feature* (spreadsheet) ability to set background-color depending on cell-value

### ISSUE THREAD

**Issue by RomaricMollard** [ONLYOFFICE/DocumentServer#243]
Can we set background-color depending on cell-value ? Like red if less than 0, or green instead ?
Currently it works on imported document from openoffice or microsoft excel but we cannot edit it (or I did not find how).
Romaric, awesome work btw :)

**Comment by ShockwaveNN** [comment]
This is called `Condition Formatting` and you're right, we already support this feature in read-only mode.
We have plans to add creating of Condition Formatting but it's not in next version

**Comment by thegruf** [comment]
Hi - Just to bump a longstanding thread - Any update on the release date for enhanced Conditional Formatting please?

We use conditional formatting extensively, so this is literally preventing our rollout out OnlyOffice more widely.
Support for working actively, not just read only, with excel basic Highlighting Cell rules would be a great start

**Comment by Rita-Bubnova** [comment]
This enhancement - issue 18072 in our private issue tracker.
But we got no ETA for conditional formatting.

**Comment by thegruf** [comment]
Hey - thanks for the quick reply.
I can only ask to please try to find some priority for this feature it would really extend the opportunity to use OnlyOffice spreadsheets.  I know I am not the only user waiting  for this feature.
Keep up the great work :)

**Comment by ShockwaveNN** [comment]
@thegruf Sorry, there is nothing we can do to change this feature priority, but you can contact sales@onlyoffice.com and discuss development priorities with sales team

**Comment by thegruf** [comment]
I absolutely understand, no doubt you have a never ending list of priorities, 
I would just like to keep this on the agenda please. I think many potential users would benefit from this feature even just as basic cell highlighting, As you can read the conditional format highlight from .xlsx it seems like maybe you have done the hard work already.
I will drop an email to sales@ and offer some encouragement there too ;-)
Appreciate the work you guys are doing tho'

**Comment by jsuhm** [comment]
Good news today:
https://www.onlyoffice.com/blog/2020/10/onlyoffice-docs-v6-0-with-big-updates-for-spreadsheets/

"**Conditional formatting for viewing**

Conditional formatting allows users to visually explore data, detect issues, and identify trends. In version 6.0 we added support for all the conditions and gradients.

Applying conditional formatting will be added in the next versions of ONLYOFFICE Docs."

But I didn't try yet.
What does "for viewing" mean exactly ? And how to use it ?

**Comment by janxkoci** [comment]
> What does "for viewing" mean exactly ? And how to use it ?

If your file already has conditional formating (e.g. made by MS Excel or LO Calc), you will be able to see it when you open the file in OnlyOffice.


Here is an example of one of my spreadsheets:
![Snímek z 2021-01-11 17-58-35](https://user-images.githubusercontent.com/19331850/104213510-a8175280-5436-11eb-8502-5e32e7df5b6d.png)

**Comment by cpot** [comment]
> 
> 
> @thegruf Sorry, there is nothing we can do to change this feature priority, but you can contact [sales@onlyoffice.com](mailto:sales@onlyoffice.com) and discuss development priorities with sales team

I still don't really understand this advice. If we contact sales and pay it goes up in the TODO list and pay even more it goes even upper in the TODO list with an ETA we can define ?  I just want to understand (cc: @galegod / Galina) 

Is this an option we can say to our customers ?  (sorry in french https://community.ceo-vision.com/topic/498/onlyoffice-mise-en-forme-conditionnelle) 

Best,

**Comment by ShockwaveNN** [comment]
> If we contact sales and pay it goes up in the TODO list and pay even more it goes even upper in the TODO list with an ETA we can define

Generally speaking I think you're right

Conditional Formatting is highly requested feature, but also highly complex as far as I'm aware, so this is not implemented in quality we want it to be

**Comment by janxkoci** [comment]
> Conditional Formatting is highly requested feature, but also highly complex as far as I'm aware, so this is not implemented in quality we want it to be

I agree it is a complex feature, but maybe it can be done in two steps:

- **Colour spectra (left)** - this should be rather easy to implement, most pieces are there, you just need to implement the rules to calculate the spectrum properly based on values, which can be done with simple macro. And then the UI. Note that _**Google Docs** have this_, while they still lack the data bars.
- **Data bars (right)** - these are more advanced and will probably require a lot of new work. These are generally supported by fullblown desktop office suits (such as MS Office, LibreOffice, WPS Office). The example file below was made with LO Calc.


![Snímek z 2021-05-04 10-51-00](https://user-images.githubusercontent.com/19331850/116980543-a5cb5380-acc6-11eb-8c45-cd30014da44b.png)

I think it would be acceptable for most users to have at least the colour spectra available. Google Docs certainly think the same.

**Comment by janxkoci** [comment]
## Update
I noticed that after I resaved the above file in OnlyOffice, it still rendered properly in OnlyOffice but not anymore in LibreOffice (where it was created). Thus OnlyOffice no longer preserves proper formatting of files after modification:

![Snímek z 2021-06-09 17-35-34](https://user-images.githubusercontent.com/19331850/121385521-78b73400-c949-11eb-875f-cf88c77646d3.png)

This is with OO `6.2.0.148`.

**Comment by ShockwaveNN** [comment]
@janxkoci Please create a new issue for this problem and attach original file before modification with ONLYOFFICE DesktopEditors

**Comment by ZeikoFr** [comment]
HI,

Is there any news on this feature (ETA or anything else) ?

Thanks

**Comment by ShockwaveNN** [comment]
@ZeikoFr We are planning to release full conditional support in next release DocumentServer v6.4

I cannot guarantee that we 100% make this version, since this feature is very complex and may be some major issues soon, but currently this feature looking good and we really hope to ship it with next release maybe in month or two
